spec/lib/good_job/job_spec.rb

```diff
@@ -204,11 +204,16 @@ def perform(result_value = nil, raise_error: false)
     end
 
     it 'can preserve the job' do
-      good_job.perform(destroy_after: false)
+      current_preserve_job_records = GoodJob.preserve_job_records
+      GoodJob.preserve_job_records = true
```

Comment:
The RSpec-y way to do this is with a [message stub](https://relishapp.com/rspec/rspec-mocks/v/3-9/docs/configuring-responses) which will be automatically cleaned up and reset at the end of the test example:  `allow(GoodJob).to receive(:preserve_job_records).and_return(true)`